### PR TITLE
Docs + Some API

### PR DESCRIPTION
**Pull request by Thundermaker300**
Also adds `Player.CurrentArmor` and `static Door.GetClosest()`

### CODE REVIEW

File: Exiled.API/Features/Door.cs [hunk 1]
```diff
@@ -289,6 +289,29 @@ public static Door Get(string name)
...
+        /// Returns the closest <see cref="Door"/> to the given <paramref name="position"/>.
+        /// </summary>
+        /// <param name="position">The <see cref="Vector3">position</see> to find the closest door to.</param>
+        /// <param name="distance">The distance between the door and the point.</param>
+        /// <returns>The door closest to the provided position.</returns>
+        public static Door GetClosest(Vector3 position, out float distance)
+        {
+            distance = 9999f;
+            Door doorToReturn = null;
+            foreach (Door door in List)
```
Comment (iopietro) [inline Exiled.API/Features/Door.cs:302]:
```suggestion
            distance = null;
            Door doorToReturn = null;
            
            foreach (Door door in List)
```

File: Exiled.API/Features/Door.cs [hunk 2]
```diff
@@ -289,6 +289,29 @@ public static Door Get(string name)
         /// <returns>The <see cref="Door"/> with the given <see cref="DoorType"/> or <see langword="null"/> if not found.</returns>
         public static Door Get(DoorType doorType) => List.FirstOrDefault(x => x.Type == doorType);
 
+        /// <summary>
+        /// Returns the closest <see cref="Door"/> to the given <paramref name="position"/>.
+        /// </summary>
+        /// <param name="position">The <see cref="Vector3">position</see> to find the closest door to.</param>
+        /// <param name="distance">The distance between the door and the point.</param>
+        /// <returns>The door closest to the provided position.</returns>
+        public static Door GetClosest(Vector3 position, out float distance)
```
Comment (iopietro) [inline Exiled.API/Features/Door.cs:298]:
```suggestion
        public static Door GetClosest(Vector3 position, out float? distance)
```

File: Exiled.API/Features/Door.cs [hunk 3]
```diff
@@ -289,6 +289,29 @@ public static Door Get(string name)
...
+        {
+            distance = 9999f;
+            Door doorToReturn = null;
+            foreach (Door door in List)
+            {
+                float doorDistance = Vector3.Distance(position, door.Position);
+                if (doorDistance < distance)
```
Comment (iopietro) [inline Exiled.API/Features/Door.cs:305]:
```suggestion
                float doorDistance = Vector3.Distance(position, door.Position);
                
                if (doorDistance < distance)
```

File: Exiled.API/Features/Player.cs [hunk 4]
```diff
@@ -405,7 +411,7 @@ public Player Cuffer
         /// </summary>
         public Vector3 Position
         {
-            get => GameObject.transform.position;
+            get => ReferenceHub.transform.position;
```
Comment (iopietro) [inline Exiled.API/Features/Player.cs:414]:
```suggestion
            get => Transform.position;
```

File: Exiled.API/Features/Player.cs [hunk 5]
```diff
@@ -415,7 +421,7 @@ public Vector3 Position
         /// <returns>Returns the direction the player is looking at.</returns>
         public Vector3 Rotation
         {
-            get => GameObject.transform.eulerAngles;
+            get => ReferenceHub.transform.eulerAngles;
```
Comment (iopietro) [inline Exiled.API/Features/Player.cs:424]:
```suggestion
            get => Transform.eulerAngles;
```

File: Exiled.API/Features/Player.cs [hunk 6]
```diff
@@ -782,6 +789,11 @@ public Item CurrentItem
             }
         }
 
+        /// <summary>
+        /// Gets the armor that the player is currently wearing. Value will be <see langword="null"/> if the player is not wearing any armor.
+        /// </summary>
+        public Armor CurrentArmor => Inventory.TryGetBodyArmor(out BodyArmor armor) ? (Armor)Item.Get(armor) : null;
```
Comment (iopietro) [inline Exiled.API/Features/Player.cs:795]: Should we move this to the HumanRole class?

**Review by iopietro — CHANGES_REQUESTED** [review]

File: Exiled.API/Features/Door.cs (hunk 1 above)
Comment (Thundermaker300) [reply Exiled.API/Features/Door.cs:302]: Won't this error? int < null

File: Exiled.API/Features/Player.cs (hunk 6 above)
Comment (Thundermaker300) [reply Exiled.API/Features/Player.cs:795]: I don't think so. The rest of the inventory stuff is in the player class, armor should be as well

File: Exiled.API/Features/Door.cs (hunk 1 above)
Comment (iopietro) [reply Exiled.API/Features/Door.cs:302]: I updated the code below.

File: Exiled.API/Features/Door.cs [hunk 7]
```diff
@@ -289,6 +289,29 @@ public static Door Get(string name)
...
+                {
+                    doorToReturn = door;
+                    distance = doorDistance;
+                }
+            }
+
+            return doorToReturn;
```
Comment (iopietro) [inline Exiled.API/Features/Door.cs:312]:
```suggestion
            Door doorToReturn = List.OrderBy(door => Vector3.Distance(position, door.Position)).FirstOrDefault()
    
            distance = Vector3.Distance(position, doorToReturn.Position);  
            return doorToReturn;
```

**Comment by iopietro** [comment]
Fix build errors.

**Comment by Thundermaker300** [comment]
> Fix build errors.

your change broke it 😭

**Pull request merged**